src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -164,6 +165,18 @@ export class DatasetService {
     });
   }
 
+  public clearCache(): void {
+    this.sessionStorage.clear();
+  }
+
+  public getLastCacheTime(): string | undefined {
+    return this.sessionStorage.get('lastCacheTime');
```

Comment:
Assuming you do use `Date.now()` this line changes to
```suggestion
    return Number(this.sessionStorage.get('lastCacheTime')) || undefined;
```

The reason I propose `Number` and not `parseInt` is that `parseInt` will try to parse the numeric part of a mixed alphanumeric string which we don't want because the value should be fully numeric, and if it is not, we need to fail to parse.

The reason for `|| undefined` is that we should return `undefined` for all invalid values: `NaN` and `0`.